doc/devel/coding_style_guideline.rst

```diff
@@ -0,0 +1,169 @@
+=============================
+DWI_ML Coding Style Guideline
+=============================
+
+The main principles behind DWI_ML development are:
+
+* **Robustness**: the results of a piece of code must be verified
+  systematically, and hence stability and robustness of the code must be
+  ensured, reducing code redundancies.
+* **Readability**: the code is written and read by humans, and it is read
+  much more frequently than it is written.
+* **Consistency**: following these guidelines will ease reading the code,
+  and will make it less error-prone.
+* **Documentation**: document the code. Documentation is essential as it helps
+  clarifying certain choices, helps avoiding obscure places, and is a way to
+  allow other members *decode* it with less effort.
+* **Language**: the code must be written in English. Norms and spelling
+  should be abided by.
+
+Coding style
+============
+
+DWI_ML uses the standard Python `PEP8`_ style to ensure the readability and
+consistency across the toolkit. Conformance to the PEP8 syntax is checked
+automatically when requesting to push to DWI_ML. There are
+`software tools <https://pypi.python.org/pypi/pep8>`_ that check your code for
+PEP8 compliance, and most text editors can be configured to check the
+compliance of your code with PEP8.
+
+You should try to ensure that your code, including comments, conform to the
+above principles.
+
+Imports
+-------
+
+DWI_ML recommends using the following package aliases to increase consistency
+and readability across the library::
+
+    import matplotlib.pyplot as plt
+    import nibabel as nib
+    import numpy as np
+    import numpy.testing as npt
+
+No alias should be used for ``h5py``::
+
+    import h5py
+
+Stick to the *common* or *recommended* import practices for each standard
+packages::
+
+    from datetime import (datetime, timedelta)
+
+It is understood as *common* or *recommended* practice one that is used in the
+documentation or examples of the package.
+
+Use generic imports for other packages::
+
+    import package
+    import package.subpackage.subsubpackage
+
+
+Naming guidelines
+-----------------
+
+The following is a set of conventions to make the DWI_ML to improve the code
+readability and make it consistent.
+
+General considerations
+^^^^^^^^^^^^^^^^^^^^^^
+
+* Keep name length within a four word maximum, and avoid gratuitous context.
+* Only use correctly-spelled dictionary words and abbreviations that appear in a
+  dictionary. Make exceptions for ``id`` and documented domain-specific
+  abbreviations (e.g. ``fa``, ``md`, ``acc``, ``pos``).
+* Names are usually read left-to-right, from more specific to less specific:
+  e.g. ``L2RegularizedSphericalDeconvolutionFilter`` instead of
+  ``SphericalDeconvolutionL2RegularizedFilter``.
+
+Package names
+^^^^^^^^^^^^^
+
+* Use a noun name: e.g. ``learning`` instead of ``learners``. An exception could
```

Comment:
Not sure I understand the sentence. Don't you mean "learners instead of learning"? And model is a noun, how is it an exception? Or then you meant "use an action name"?

Reply:
> Don't you mean "learners instead of learning"?

Right the opposite: the package name should be e.g. `learning` instead of `learners`, or `filtering` instead of `filters`. I can change it to `action name` to make it clearer.

Reply:
Made it more clear in ad7ee3c.